genesis/process/intermediate/delegationProcessor.go

```diff
@@ -86,20 +136,103 @@ func (dp *delegationProcessor) getDelegationScOnCurrentShard() ([]genesis.Initia
 		}
 	}
 
+	log.Trace("getDelegationScOnCurrentShard",
+		"num delegation SC", len(smartContracts),
+		"shard ID", dp.shardCoordinator.SelfId(),
+	)
 	return smartContracts, nil
 }
 
+func (dp *delegationProcessor) setDelegationStartParameters(smartContracts []genesis.InitialSmartContractHandler) error {
+	for _, sc := range smartContracts {
+		delegatedNodes := dp.nodesListSplitter.GetDelegatedNodes(sc.AddressBytes())
+		numNodes := len(delegatedNodes)
+
+		log.Trace("setDelegationStartParameters",
+			"SC owner", sc.GetOwner(),
+			"SC address", sc.Address(),
+			"num delegated nodes", numNodes,
+			"node price", dp.nodePrice.String(),
+			"shard ID", dp.shardCoordinator.SelfId(),
+		)
+
+		err := dp.executeSetNumNodes(numNodes, sc)
+		if err != nil {
+			return err
+		}
+
+		err = dp.executeSetNodePrice(sc)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (dp *delegationProcessor) executeSetNumNodes(numNodes int, sc genesis.InitialSmartContractHandler) error {
+	setNumNodesTxData := fmt.Sprintf("%s@%x", setNumNodesFunction, numNodes)
```

Comment:
maybe use strings and plus, instead of fmt.sprintf.

Reply:
I prefer to use fmt.Sprintf where I have more than one addition. I also have a %x and this should get uglier